bot/cogs/bigbrother.py

```diff
@@ -84,16 +91,26 @@ def update_cache(self, api_response: List[dict]):
             infraction_list = await response.json()
         except ClientError:
             log.exception(f"Failed to retrieve bb watch reason for {user_id}.")
-            return "(error retrieving bb reason)"
+            return WatchInformation(reason="(error retrieving bb reason)", actor_id=None, inserted_at=None)
 
         if infraction_list:
+            # Get the latest watch reason
             latest_reason_infraction = max(infraction_list, key=self._parse_infraction_time)
+
+            # Get the actor of the watch/nominate action
+            actor_id = int(latest_reason_infraction["actor"]["user_id"])
+
+            # Get the date the watch was set
+            date = latest_reason_infraction["inserted_at"]
+
+            # Get the latest reason without the prefix
             latest_reason = latest_reason_infraction['reason'][len(self.infraction_watch_prefix):]
+
             log.trace(f"The latest bb watch reason for {user_id}: {latest_reason}")
-            return latest_reason
+            return WatchInformation(reason=latest_reason, actor_id=actor_id, inserted_at=date)
 
-        log.trace(f"No bb watch reason found for {user_id}; returning default string")
-        return "(no reason specified)"
+        log.trace(f"No bb watch reason found for {user_id}; returning defaults")
+        return WatchInformation(reason="(no reason specified)", actor_id=None, inserted_at=None)
 
     @staticmethod
     def _parse_infraction_time(infraction: str) -> struct_time:
```

Comment:
I'm not sure why this hasn't errored out before, but it errors for me when testing.

We've got `infraction` typed as `str` and then attempt to index it as a dictionary:

https://github.com/python-discord/bot/blob/0f501deddfc1ec5bf82b5bfdc1dc07dfd63b1a8e/bot/cogs/bigbrother.py#L115-L120

Reply:
Oh, that's an oversight from me; the type annotation should say dictionary. I'll also update the docstring to reflect that. 

It's basically a helper function to avoid a longer inline lambda to be able to sort the infraction dictionaries based on a converted key.

Reply:
Done.